From 78eb1b722a116192ff42335ada461cff6e4e4614 Mon Sep 17 00:00:00 2001
From: Dinolek <git@dinolek.me>
Date: Sun, 6 Jul 2025 20:19:12 +0200
Subject: [PATCH 112/115] drm/mediatek: dsi: Clear DSI_INTSTA before requesting
 irq

This is needed for cases where bootloader didn't clear it, which leads to
handler being stuck.

Signed-off-by: Dinolek <git@dinolek.me>
---
 drivers/gpu/drm/mediatek/mtk_dsi.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/mediatek/mtk_dsi.c b/drivers/gpu/drm/mediatek/mtk_dsi.c
index 49dfc0f82..795adcd2e 100644
--- a/drivers/gpu/drm/mediatek/mtk_dsi.c
+++ b/drivers/gpu/drm/mediatek/mtk_dsi.c
@@ -1251,6 +1251,7 @@ static int mtk_dsi_probe(struct platform_device *pdev)
 	if (ret < 0)
 		return dev_err_probe(dev, ret, "Failed to register DSI host\n");
 
+	writel(0, dsi->regs + DSI_INTSTA);
 	ret = devm_request_irq(&pdev->dev, irq_num, mtk_dsi_irq,
 			       IRQF_TRIGGER_NONE, dev_name(&pdev->dev), dsi);
 	if (ret) {
-- 
2.56.0

